### PR TITLE
LUN-2160: Change "Redirect" field from CharField to URLField

### PR DESCRIPTION
**Pull request by bogdan-cornianu**
Change "Redirect" field from CharField to URLField in order to apply URL validation.

### CODE REVIEW

File: cms/admin/forms.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ class PageForm(PageAddForm):
     # moderation - message is a fake field
     moderator_message = forms.CharField(max_length=1000, widget=forms.HiddenInput, required=False)
     
-    redirect = forms.CharField(label=_('Redirect'), max_length=255, required=False,
+    redirect = forms.URLField(label=_('Redirect'), max_length=255, required=False,
```
Comment (earthboundkid) [inline cms/admin/forms.py:169]: Do `URLFields` let you do relative URLs like `/target-page/` or only absolute URLs like `http://example.com/target-page/`?

Comment (bogdan-cornianu) [inline cms/admin/forms.py:169]: You can only use absolute URLs.

Comment (earthboundkid) [inline cms/admin/forms.py:169]: Redirecting to a relative URL is valid as far as HTTP goes, and there are already many relative redirects saved in the database. I don't think we should make this change.

Comment (earthboundkid) [inline cms/admin/forms.py:169]: I propose instead that you write a `clean_redirect` method that rejects URLs if it contains whitespace or doesn't start with `http://`, `https://`, `/`, or `../`.

Comment (earthboundkid) [inline cms/admin/forms.py:169]: The regex `r'^(https?://|/|../)\S+$'` matches our basic criteria for good URLs, and `urlparse.urlparse("http://exam]ple.com")` raises `ValueError: Invalid IPv6 URL`, which fixes the other problem mentioned in the ticket.

**Comment by bogdan-cornianu** [comment]
I'm going to close this pull request since there is a better place to validate the redirect field: lunchbox / admin / cms_extensions.py

**Pull request closed without merging**